modules/client-configuration/pages/contact-methods-saltssh-tunnel.adoc

```diff
@@ -0,0 +1,97 @@
+[[contact-methods-pushssh]]
+= SSH Push (with tunnel) contact method
+
+
+SSH Push (with tunnel) ([literal]``ssh-push-tunnel``) is used in environments where clients cannot reach the {productname} Server directly.
+In this environment, clients are located in a firewall-protected zone called a DMZ.
+No system within the DMZ is authorized to open a connection to the internal network, including the {productname} Server.
+
+This SSH method creates an encrypted tunnel from the {productname} Server on the internal network to the clients located on the DMZ.
+After all actions and events are executed, the tunnel is closed.
+
+The server uses SSH to contact the clients at regular intervals, checking in and performing scheduled actions and events.
+
+
+// FiXME: only here with tunnel, otherwise move to contact-methods-saltssh.adoc
+[IMPORTANT]
+====
+Re-installing systems using the provisioning model is not currently supported on clients managed with SSH Push.
+====
+
+
+[IMPORTANT]
+====
+The tunnel is used to provide the access to the server through the encrypted tunnel.
+The repositories assigned to the SSH Push client (with tunnel) are provided through this tunnel only, thus it is not possible to use package manager tools directly from the client system because the repositories are available only while the tunnel is up.
+In other words, access is only possible if the session is initiated by the server.
+All package managing operations on the client can be performed from the server side only.
+====
+
+For tunneling connections via SSH, a port number is required for tunneling via HTTPS.
+The port number used by default is `1233`.
+To overwrite it, you can add a custom port numbers greater than 1024 to [path]``/etc/rhn/rhn.conf``:
+
+----
+ssh_push_port_https = high_port
+----
+
+
+For security reasons, you might want to use sudo with SSH, to access the system as an unprivileged user instead of as root.
+
+
+.Procedure: Configuring Unprivileged SSH Access
+. Ensure you have the latest [path]``spacewalk-taskomatic`` and [path]``spacewalk-certs-tools`` packages installed on the {productname} Server.
+. On each client system, create an appropriate unprivileged user.
+. On each client system, edit the [filename]``sudoers`` file:
++
+----
+sudo visudo
+----
+. Grant [command]``sudo`` access to the user by adding this line at the end of the [filename]``sudoers`` file.
+  Replace [systemitem]``<user>`` with the name of the user that is bootstrapping the client in the {webui}:
++
+----
+<user>  ALL=NOPASSWD: /usr/bin/python, /usr/bin/python2, /usr/bin/python3, /var/tmp/venv-salt-minion/bin/python
```

Comment:
I would suggest to exclude at least python 2 related pathes from the list. Actually we support Salt-SSH with the salt bundle only now, am I right? If so, only `/var/tmp/venv-salt-minion/bin/python` is really required here, the others are possible, but only with extra tweaks on the server side to allow salt-ssh without the bundle, but as we've dropped already py2x-compat-salt packages, salt-ssh without the bundle could work with python 3 only.

Reply:
ok, I'll remove python and python2, and will add a reference to salt bundle (maybe, as a separate PR).